include/powsybl/iidm/ExtensionProviders.hxx

```diff
@@ -46,10 +71,53 @@ const T& ExtensionProviders<T, Dummy>::findProviderOrThrowException(const std::s
 }
 
 template <typename T, typename Dummy>
-void ExtensionProviders<T, Dummy>::registerExtension(const std::string& name, std::unique_ptr<T>&& provider) {
-    getInstance().m_providers.emplace(std::make_pair(name, std::move(provider)));
+std::vector<std::string> ExtensionProviders<T, Dummy>::getFiles(const std::string& directory, const boost::regex& file) {
+    std::vector<std::string> libs;
+    if (boost::filesystem::exists(directory) && boost::filesystem::is_directory(directory)) {
+        boost::filesystem::directory_iterator it(directory);
+        boost::filesystem::directory_iterator endit;
+
+        while (it != endit)
+        {
+            if (boost::filesystem::is_regular_file(*it) && boost::regex_match((*it).path().filename().c_str(), file)) {
+                libs.push_back((directory / it->path().filename()).string());
+            }
+            ++it;
+        }
+    }
+
+    return libs;
+}
+
+template <typename T, typename Dummy>
+ExtensionProviders<T, Dummy>& ExtensionProviders<T, Dummy>::getInstance() {
+    static ExtensionProviders<T, Dummy> s_instance;
+    return s_instance;
 }
 
+template <typename T, typename Dummy>
+void ExtensionProviders<T, Dummy>::loadLibrary(const std::string& library) {
+    logging::Logger& logger = logging::LoggerFactory::getLogger<ExtensionProviders>();
+    boost::dll::shared_library lib(library);
+
+    if (m_loadedLibraries.find(boost::filesystem::canonical(lib.location())) == m_loadedLibraries.end()) {
+        if (lib.has("createSerializers")) {
+            const auto &createSerializers = boost::dll::import_alias<std::map<std::string, std::unique_ptr<T>>()>(lib, "createSerializers");
+            for (auto &it : createSerializers()) {
+                const auto &status = m_providers.emplace(std::make_pair(it.first, std::move(it.second)));
+
+                // check that extension was not already registered
+                if (!status.second) {
+                    throw PowsyblException(logging::format("Extension %1% was already registered", it.first));
+                }
+                logger.info(logging::format("Loaded extension %1% from %2%", it.first, library));
```

Comment:
I think the message is not correct:
Extension X has been loaded from Y

Maybe you should change the level to debug